### PR TITLE
Remove ethereum addresses from @livepeer/player

### PR DESCRIPTION
**Pull request by iameli**
**What does this pull request do? Explain your changes. (required)**
`@livepeer/player` shan't use Ethereum addresses going forward. This PR removes them and has all pertinent logic utilize HLS manifest URLs directly instead.

**Specific updates (required)**
- A few changes to get player building again, see https://github.com/livepeer/livepeerjs/pull/316/files
- Added a step that builds `player` and `explorer` in the CI process so that we get more warning if the build is broken for either of them
- Lots of GraphQL/Apollo/Web3 removal
- A new container class, `<QueryHandler>`, that allows for shared logic between `<Channel>` (which is kinda poorly named now) and `<Embed>`.
- Some routing tweaks to implement the above
- Misc polish all over; added `<form>` tags for searching where appropriate

**How did you test each of these updates (required)**
- Ran through the Landing, Channel, and Embed views, attempting to utilize every feature
- Checked to make sure video still played on each view, checked to make sure that live --> dead and dead --> live transitions work okay

**Does this pull request close any open issues?**

**Screenshots (optional):**
image
image
image
image

### CODE REVIEW

**Comment by iameli** [comment]
Hmm, the remaining test failure is codeclimate complaining about files in `explorer` that I didn't touch. I'm happy to refactor, but presumably that shouldn't block this merge?

**Comment by iameli** [comment]
And upon a second glance, both things flagged by codeclimate are destructuring assignments that would be difficult to refactor away in any case.

**Comment by Randy1Burrell** [comment]
@iameli In response to your question, no that would not block this merge once everything is good with your PR otherwise. I will review this PR tomorrow.

**Comment by Randy1Burrell** [comment]
@iameli great work so far. I am getting this error whenever I test the explorer on small screen and try to type in the search bar that appears in the dropdown menu.

```JavaScript
Unhandled Rejection (TypeError): Cannot read property 'item' of undefined
_callee$
src/components/BasicNavbar.js:197
  194 |   </Btn>
  195 | }
  196 | onSelect={async ({ detail }) => {
> 197 |   const { action } = detail.item.dataset
      | ^  198 |   switch (action) {
  199 |     case 'feedback':
  200 |       return window.open(
View compiled
▶ 7 stack frames were collapsed.
(anonymous function)
src/components/BasicNavbar.js:196
  193 |     <Icon use="more_vert" />
  194 |   </Btn>
  195 | }
> 196 | onSelect={async ({ detail }) => {
      | ^  197 |   const { action } = detail.item.dataset
  198 |   switch (action) {
  199 |     case 'feedback':
View compiled
▼ 18 stack frames were expanded.
HTMLUnknownElement.callCallback
node_modules/react-dom/cjs/react-dom.development.js:147
invokeGuardedCallbackDev
node_modules/react-dom/cjs/react-dom.development.js:196
invokeGuardedCallback
node_modules/react-dom/cjs/react-dom.development.js:250
invokeGuardedCallbackAndCatchFirstError
node_modules/react-dom/cjs/react-dom.development.js:265
executeDispatch
node_modules/react-dom/cjs/react-dom.development.js:571
executeDispatchesInOrder
node_modules/react-dom/cjs/react-dom.development.js:596
executeDispatchesAndRelease
node_modules/react-dom/cjs/react-dom.development.js:695
executeDispatchesAndReleaseTopLevel
node_modules/react-dom/cjs/react-dom.development.js:704
forEachAccumulated
node_modules/react-dom/cjs/react-dom.development.js:674
runEventsInBatch
node_modules/react-dom/cjs/react-dom.development.js:844
runExtractedEventsInBatch
node_modules/react-dom/cjs/react-dom.development.js:852
handleTopLevel
node_modules/react-dom/cjs/react-dom.development.js:5025
batchedUpdates$1
node_modules/react-dom/cjs/react-dom.development.js:19916
batchedUpdates
node_modules/react-dom/cjs/react-dom.development.js:2246
dispatchEvent
node_modules/react-dom/cjs/react-dom.development.js:5105
interactiveUpdates$1
node_modules/react-dom/cjs/react-dom.development.js:19978
interactiveUpdates
node_modules/react-dom/cjs/react-dom.development.js:2267
dispatchInteractiveEvent
node_modules/react-dom/cjs/react-dom.development.js:5081
▲ 18 stack frames were expanded.
This screen is visible only in development. It will not appear if the app crashes in production.
Open your browser’s developer console to further inspect this error.
```

image

image

**Comment by iameli** [comment]
@Randy1Burrell Thanks for the review — addressed in e482f8c32ce276e85bdfb9feb0abcef65ddae7e5

**Comment by iameli** [comment]
Rebased to clean up some excessive commits and incorporate removing `iconOptions`, which React was complaining about.

File: packages/player/src/containers/App.js [hunk 1]
```diff
@@ -3,13 +3,26 @@ import { Redirect, Route, Switch } from 'react-router-dom'
 import Landing from '../views/Landing'
 import Channel from '../views/Channel'
 import Embed from '../views/Embed'
+import QueryHandler from './QueryHandler'
+
+// https://player.livepeer.org/play?url=https://abc123.cloudflare.net/something/current
```
Comment (Randy1Burrell) [inline packages/player/src/containers/App.js]: Can we remove this if it's not being used anymore?

File: packages/player/src/containers/App.js [hunk 2]
```diff
@@ -3,13 +3,26 @@ import { Redirect, Route, Switch } from 'react-router-dom'
...
 
 const App = ({ location }) => (
   <div>
     <Switch>
       <Route exact path="/" component={Landing} />
-      <Route exact path="/channels/:channel" component={Channel} />
-      <Route exact path="/embed/:channel" component={Embed} />
+      {/* <Route exact path="/channels/:channel" component={Channel} />
```
Comment (Randy1Burrell) [inline packages/player/src/containers/App.js]: Remove these too.

File: packages/player/src/services/index.js [hunk 3]
```diff
@@ -1,13 +1,11 @@
 import { combineReducers } from 'redux'
-import { fork } from 'redux-saga/effects'
+// import { fork } from 'redux-saga/effects'
```
Comment (Randy1Burrell) [inline packages/player/src/services/index.js]: This can be removed too.

File: packages/player/src/services/index.js [hunk 4]
```diff
@@ -1,13 +1,11 @@
...
 import * as routing from './routing'
-import * as jobs from './jobs'
 
 export const rootReducer = combineReducers({
   routing: routing.reducer,
-  jobs: jobs.reducer,
 })
 
 export function* rootSaga() {
```
Comment (Randy1Burrell) [inline packages/player/src/services/index.js]: This function doesn't make sense having, because it does nothing anymore.

**Review by Randy1Burrell — CHANGES_REQUESTED** [review]

**Comment by Randy1Burrell** [comment]
The banner at the bottom is being cut off.
image

**Review by Randy1Burrell — APPROVED** [review]
LGTM!

**Pull request merged**